pytket/pytket/wasm/wasm.py

```diff
@@ -64,169 +72,181 @@ def __init__(self, filepath: str, check_file: bool = True, int_size: int = 32):
-
-        with open(self._filepath, "rb") as file:
-            self._wasm_file: bytes = file.read()
-
-        self._wasm_file_encoded = base64.b64encode(self._wasm_file)
-
-        self._wasmfileuid = hashlib.sha256(self._wasm_file_encoded).hexdigest()
-
-        self._check_file = check_file
-
         # stores the names of the functions mapped
         #  to the number of parameters and the number of return values
         self._functions = dict()
 
         # contains the list of functions that are not allowed
-        # to use in pytket (because of types that are not i32)
-        self._unsupported_function = []
-
-        if self._check_file:
-            mod_iter = iter(decode_module(self._wasm_file))
-            _, _ = next(mod_iter)
-
-            for _, cur_sec_data in mod_iter:
-                # read in list of function signatures
-                if cur_sec_data.id == SEC_TYPE:
-                    for idx, entry in enumerate(cur_sec_data.payload.entries):
-                        function_signatures.append({})
-                        function_signatures[idx]["parameter_types"] = [
-                            self.type_lookup[pt] for pt in entry.param_types
-                        ]
-                        if entry.return_count > 1:
-                            if (
-                                isinstance(entry.return_type, list)
-                                and len(entry.return_type) == entry.return_count
-                            ):
-                                function_signatures[idx]["return_types"] = [
-                                    self.type_lookup[rt] for rt in entry.return_type
-                                ]
-                            elif isinstance(entry.return_type, int):
-                                function_signatures[idx]["return_types"] = [
-                                    self.type_lookup[entry.return_type]
-                                ] * entry.return_count
-                            else:
-                                raise ValueError(
-                                    f"Only parameter and return values of "
-                                    + f"i{self._int_size} types are"
-                                    + f" allowed, found type: {entry.return_type}"
-                                )
-                        elif entry.return_count == 1:
+        # to use in pytket (because of types that are not integers
+        # of the supplied int_size.)
+        self._unsupported_functions = []
+
+        self._wasm_module = wasm_module
+        self.checked = False
+
+        if check:
+            self.check()
+
+    def check(self) -> None:
+        """Collect functions from the module that can be used with pytket.
+
+        Populates the internal list of supported and unsupported functions
+        and marks the module as checked so that subsequent checking is not
+        required.
+        """
```

Comment:
Do we maybe want to make sure this has not run before?
Maybe not run it if `self.checked` is already true